### PR TITLE
Git Bash

### PR DESCRIPTION
**Pull request by chipshort**
I added git bash as embedded terminal.
Also, I changed to handling of working directory, to reduce the lag when starting FlashDevelop with the plugin enabled.

### CODE REVIEW

File: Plugins/DSLCompletion/DSLComplete.cs [hunk 1]
```diff
@@ -255,6 +255,8 @@ private string FindSelection(Check isAllowedChar, bool checkRight = true)
             
             var text = sci.GetLine(sci.CurrentLine);
 
+            if (!text.Contains("\"")) return null;
```
Comment (Neverbirth) [inline Plugins/DSLCompletion/DSLComplete.cs]: Why is this change in this PR?

File: Plugins/ConsolePanel/Settings.cs [hunk 2]
```diff
@@ -34,5 +40,51 @@ public ConsoleColor ForegroundColor
...
+            {
+                return this.minttyCmd;
+            }
+            set
+            {
+                this.minttyCmd = value;
+            }
+        }
+
+        [Category("Bash"), DisplayName("Parameters"), DefaultValue("-e \"/c/Program Files/Git/bin/bash\"")]
```
Comment (Neverbirth) [inline Plugins/ConsolePanel/Settings.cs]: It would be safer to use paths like %PROGRAMFILES%

File: Plugins/DSLCompletion/DSLComplete.cs (hunk 1 above)
Comment (chipshort) [inline Plugins/DSLCompletion/DSLComplete.cs]: Yeah, I just saw it myself. I did it yesterday and forgot about it.
It has nothing to do with the console plugin.

File: Plugins/ConsoleControl/ConsoleProvider.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace ConsoleControl
+{
+    public interface ConsoleProvider : System.Windows.Forms.IContainerControl
```
Comment (Neverbirth) [inline Plugins/ConsoleControl/ConsoleProvider.cs]: In .NET the common convention is to start interface names with I.

Comment (chipshort) [inline Plugins/ConsoleControl/ConsoleProvider.cs]: You are right. I will change this stuff.
Do you want me to remove the DSL commit for this PR?

File: Plugins/ConsolePanel/Settings.cs (hunk 2 above)
Comment (chipshort) [inline Plugins/ConsolePanel/Settings.cs]: I cannot do it for this specific line. As you can see it is not standard command line format. It cannot contain the ":". However, I will do it for the other one.

Comment (chipshort) [inline Plugins/ConsolePanel/Settings.cs]: Also, %PROGRAMFILES% points to "C:\Program Files (x86)" when you run FlashDevelop x86, so it does not work.

**Pull request merged**